messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -1048,11 +1048,10 @@ public void run() {
 
                     // When in an initial stage, split segments to the requested number.
                     if (tokenStoreCurrentSegments.length == 0 && segmentsSize > 0) {
-                        tokenStoreCurrentSegments = transactionManager.fetchInTransaction(
```

Comment:
Nit: the `tokenStoreCurrentSegments` can be moved inside the try-catch block right now if I see this correctly.